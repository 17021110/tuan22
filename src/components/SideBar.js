import React from 'react';
import PropTypes from 'prop-types';
import Item from './Items';
import './style.css';

SideBar.propTypes = {
  listItems: PropTypes.array,
};
SideBar.defaultProps = {
  listItems: [],
};

export default function SideBar(props) {
  const {listItems} = props;
  return (
    <div className="listcss">
      {listItems.map(
          (item, index) =>
            item.show && <Item item={item} key={index} source="list" />,
      )}
    </div>
  );
}

