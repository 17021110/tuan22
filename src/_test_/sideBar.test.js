
import React from 'react';
import { render } from '@testing-library/react';

import SideBar from '../components/SideBar';
test('should render sidebar', () =>{
  const wrapper = render(<SideBar/>);
  expect(wrapper.container).toMatchSnapshot();
})