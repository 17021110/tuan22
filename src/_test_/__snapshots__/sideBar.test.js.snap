// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homepage 1`] = `
<div>
  <div
    class="listcss"
  />
</div>
`;

exports[`should render sidebar 1`] = `
<div>
  <div
    class="listcss"
  />
</div>
`;
